src/crimson/admin/osd_admin.cc

```diff
@@ -208,8 +209,41 @@ class SeastarMetricsHook : public AdminSocketHook {
    for (const auto& [full_name, metric_family]: seastar::scollectd::get_value_map()) {
      for (const auto& [labels, metric] : metric_family) {
        if (metric && metric->is_enabled()) {
-         auto& metric_function = metric->get_function();
-         f->dump_float(metric->get_id().full_name(), metric_function().d());
+	 switch (auto v = (*metric)(); v.type()) {
+	 case data_type::GAUGE:
+	   f->dump_float(full_name, v.d());
+	   break;
+	 case data_type::COUNTER:
+	   [[fallthrough]];
+	 case data_type::DERIVE:
+	   f->dump_unsigned(full_name, v.ui());
+	   break;
+	 case data_type::HISTOGRAM: {
```

Comment:
Perhaps we should move all the code under this case to a dedicated function

Reply:
I mean under `data_type::HISTOGRAM`

Reply:
@AmnonHanuhov i moved the whole switch-case block into a dedicated function. does it look better now?

Reply:
moving the stuff under `data_type::HISTOGRAM` requires us to expose the lower-level types in seastar in our function signature, i'd like avoid doing so unless necessary.

Reply:
Thanks, looks better now